pylint/checkers/dunder_methods.py

```diff
@@ -143,6 +147,12 @@ class DunderCallChecker(BaseChecker):
     }
     options = ()
 
+    def open(self) -> None:
+        self._dunder_methods: dict[str, str] = {}
+        for since_vers, dunder_methods in DUNDER_METHODS.items():
+            if since_vers <= self.linter.config.py_version:
+                self._dunder_methods.update(dunder_methods)
```

Comment:
```suggestion
        self._dunder_methods: dict[str, str] = {
            dunder_method: text
            for since_vers, dunder_methods in DUNDER_METHODS.items():
            for dunder_method, text in dunder_methods.items()
            if since_vers <= self.linter.config.py_version:
        }
```
I think this is better performance-wise ?

Reply:
Are you sure? This seems a bit like pre-mature optimisation with much less readable code. I'd prefer the current code.

Reply:
Well, actual benchmark with the current list show that the for loop is faster by an order of magnitude.
Code:
```python
tick = 10000
start = time.time()
for i in range(tick):
    dunder_methods: dict[str, str] = {}
    for since_vers, dunder_methods in DUNDER_METHODS.items():
        if since_vers <= (3, 10):
            dunder_methods.update(dunder_methods)
end = time.time()
print(end - start)
start = time.time()
for i in range(tick):
    dunder_methods: dict[str, str] = {
        dunder_method: text
        for since_vers, dunder_methods in DUNDER_METHODS.items()
        for dunder_method, text in dunder_methods.items()
        if since_vers <= (3, 10)
    }
end = time.time()
print(end - start)
```

Result:
```
0.014837980270385742
0.13480186462402344
```
dict comprehension would be faster but for 1000+ elements which is not going to happen.